### DISCUSSION TITLE
Wait for computed property

### DISCUSSION THREAD

**Discussion by marcioelias** [livewire/livewire#5368] [1. Help] [answered]
Hello guys, I'm in a situation that I can't figure out a way to solve.

I have a select with states, when the user selects a state, a computed property "getCitiesProperty" return all cities related to the selected state, which works well.

But when loading data to edit, I need to wait for getCitiesProperty to be finished before setting the city code to the select that has a list of cities. How I can do this?

Anyone has done something similar? Any ideas on how to solve this?

**Comment by joshhanley** [comment]
@marcioelias not sure what you mean by wait for `getCitiesProperty` to be finished as it should wait already, as that's just how php works, it will only process one thing at a time. If it's not working for you, can you share some of your code that demonstrates the issue so we can have a look? Thanks!

**Comment by marcioelias** [comment]
Hello @joshhanley, see the picture above:

![image](https://user-images.githubusercontent.com/2084051/202741730-e9a9b3b5-7684-4901-af92-3df7707ef6d5.png)

- Estado = states
- Cidade = cities

On the database, I have a one-to-many relationship, where one state has many cities.

When I mount the component, the state's property has filled with all states in the database. When the user chooses a State, I need to fill in all cities related to the selected state to select representing the cities, allowing the user to select a city from this list.

In order to achieve this, I created a computed property getCitiesProperty:

```
public function getCitiesProperty()
{
    return Cities::where('state_id', $this->state_id)->get();
}
```

This works fine when I'm creating a new record, but when I need to edit an existing one, is that the problem happens.

In the mount() method I set the state_id and the city_id, but, the city_id doesn't exist on the UI yet, because the computed property has not finished populating the user interface, as a result, the city that shows on the form (selected one), is the first city on the cities list, not that one that was set from the customer model on the mount method.

I hope that I can explain better the problem now. Sorry for my English.

**Comment by joshhanley** [comment, marked as answer]
@marcioelias ah ok, makes sense. I would just populate the state and city in the mount method, and then call `$this->cities;` in mount, which should load all the values into the select dropdown for you with the correct state and city selected.

If that doesn't work, can you share your component class and view and we can see if we can help you get it running?

Hope this helps!

**Comment by marcioelias** [comment]
@joshhanley that works! Perfect, thank you for your help.

**Reply by joshhanley** [reply]
No worries! 🙂
